### PR TITLE
Synapse Homepage component

### PR DESCRIPTION
**Pull request by nickgros**
* Add component for Synapse homepage
* Temporarily add decorative images until I get S3 access
* Update the demo ProjectViewCarousel's entity ID to use the current version of the Synapse homepage Project View, so users can get live feedback when trying to put their project on the homepage.

### CODE REVIEW

File: src/lib/containers/SynapseHomepage.tsx [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+import Java from '../assets/icons/Java'
+import ProjectViewCarousel from './home_page/project_view_carousel/ProjectViewCarousel'
+import {
+  getEndpoint,
+  BackendDestinationEnum,
+} from '../utils/functions/getEndpoint'
+import { SynapseClient } from '../utils'
+
+export type SynapseHomepageProps = {
+  token?: string
```
Comment (nickgros) [inline src/lib/containers/SynapseHomepage.tsx:20]: entityId for project view should be a prop

File: src/lib/containers/SynapseHomepage.tsx [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+  const PROJECT_VIEW_ENTITY_ID = 'syn23593547.1'
+
+  const [dashboardLink, setDashboardLink] = useState('')
+
+  useEffect(() => {
+    const getDashboardLink = async () => {
+      if (token) {
+        const userProfile = await SynapseClient.getUserProfile(token)
+        setDashboardLink(
+          `${getEndpoint(BackendDestinationEnum.PORTAL_ENDPOINT)}#!Profile:${
```
Comment (nickgros) [inline src/lib/containers/SynapseHomepage.tsx:43]: can replace id with `v`, then no need to make a web request, no need for useEffect

File: src/lib/containers/SynapseHomepage.tsx [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+        <div className="FlexContainer">
+          <DataSharingGraphic className="SectionGraphic" />
+          <div className="SectionTextFlexContainer">
+            <h2>Our Data Sharing Principles</h2>
+            <div>
+              <p className="SectionBody">
+                Synapse operates under a complete governance process that
+                includes well-documented Terms and Conditions of Use, guidelines
+                and operating procedures, privacy enhancing technologies, as
+                well as the right of audit and external reviews.{' '}
```
Comment (nickgros) [inline src/lib/containers/SynapseHomepage.tsx:231]: drop the {' '}

File: src/lib/containers/SynapseHomepage.tsx [hunk 4]
```diff
@@ -0,0 +1,463 @@
...
+              href="https://python-docs.synapse.org/build/html/CommandLineClient.html"
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              <Terminal />
+              <span>Command Line Client</span>
+            </a>
+            <a
+              className="ClientFlexItem"
+              href="https://github.com/Sage-Bionetworks/Synapse-Repository-Services/blob/develop/client/synapseJavaClient/src/main/java/org/sagebionetworks/client/SynapseClient.java"
```
Comment (nickgros) [inline src/lib/containers/SynapseHomepage.tsx:281]: open a ticket about linking to github and how we treat the java client

File: src/lib/containers/SynapseHomepage.tsx [hunk 5]
```diff
@@ -0,0 +1,463 @@
...
+              <p className="SectionBody">
+                Create projects, upload &amp; download files, generate
+                provenance, query, create wikis and more all from the comfort of
+                your own code.
+              </p>
+            </div>
+            <Button
+              onClick={() =>
+                window.open(
+                  'https://docs.synapse.org/articles/api_documentation.html',
```
Comment (nickgros) [inline src/lib/containers/SynapseHomepage.tsx:301]: perhaps this page should be what the iconography and links reflect

File: src/lib/containers/SynapseHomepage.tsx [hunk 6]
```diff
@@ -0,0 +1,463 @@
...
+                />
+              </a>
+            </div>
+          </div>
+        </div>
+      </div>
+      <div className="SynapseHomepage__Section PrimaryBackground">
+        <div className="SynapseHomepage__Section__Centered">
+          <div
+            style={{
```
Comment (nickgros) [inline src/lib/containers/SynapseHomepage.tsx:420]: Pull explicit styling out to css

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**